name: Monitor and Create Issue on Workflow Failure

on:
  workflow_run:
    workflows: ["python-app.yml"]
    types:
      - completed

jobs:
  check_and_create_issue:
    runs-on: ubuntu-latest
    steps:
      - name: Check Workflow Status
        id: check_status
        run: echo "Workflow status is ${{ github.event.workflow_run.conclusion }}"
      
      - name: Create Issue
        if: ${{ steps.check_status.outputs.check_status != 'success' }}
        uses: actions/github-script@v5
        with:
          script: |
            const branchName = github.event.workflow_run.head_ref;
            const initiator = github.event.workflow_run.actor;

            const issueTitle = `Failure in ${branchName} initiated by ${initiator}`;
            const issueBody = `The workflow 'python-app.yml' failed in branch ${branchName}.`;
            
            const octokit = new github.GitHub(github.token);
            await octokit.issues.create({
              owner: github.context.repo.owner,
              repo: github.context.repo.repo,
              title: issueTitle,
              body: issueBody,
              assignees: ["Teebra", "cerrybin"] // Assign the issue to specific users
            });
